### ISSUE TITLE
Push series to trakt

### ISSUE THREAD

**Issue by Js41637** [SchizoDuckie/DuckieTV#459]
This still needs to be added back in. I have been doing some research on this and it looks a bit confusing, also states it requires OAuth so I doubt it would work with our Token based Authentication which is getting deprecated soon anyway and is why is why our Trakt Sync doesn't work.. we think.

Anyway. There seems to be two ways to push series to TraktTV. There is `add to collection` which will add individual episodes or whole shows if only the show is passed or season, ect.

The other method is `add to history`. The main difference here is adding to history marks the episode/season/show as watched with an optional watched_at if specified. Add to collection just marks the show as being 'owned'. And it works in the same way as Add to Collection.

The problem however is Adding to History and Adding to Collection are two separate things, marking a show as watched adds it to the users history on TraktTV and our 'Sync from TraktTV' only fetches a users Collection, not their history. So marking shows as watched will not be returned when we try and Sync a users collection from TraktTV.

So similar to when we Pull from Trakt we will need to push all series to the collection and then push all watched episodes/series so that when someone Pulls their shows from trakttv again, it will retain their shows marked as watched.

http://docs.trakt.apiary.io/#reference/sync/add-to-history/add-items-to-watched-history
http://docs.trakt.apiary.io/#reference/sync/add-to-collection/add-items-to-collection

**Comment by garfield69** [comment]
So I take it that adding a show to history does not automatically add it to
the collection? Seems like trakt.tv could have made that obvious feature
available...


  _____

**Comment by garfield69** [comment]
related #336

**Comment by Js41637** [comment]
Yes adding a show to history does not add it to the collection. I only quickly tested with a single episode marking it as watched and it wasn't returned when I fetched my Collection. I think it's safe to assume it's the same for whole series.

Although seeing as we can add whole series at once, I don't think it would be hard to just send all the series to the collection and then send all the watched episodes.

**Comment by SchizoDuckie** [comment]
Looks like this works. i just pushed everything in my library

**Comment by lonelyfairie** [comment]
Just wanted to report that on mac using the latest angular version this worked correctly, I pushed my library a couple days ago and just now marked a day in the calendar as watched and can confirm that it pushed that to trakt without any additional user interaction.

**Comment by garfield69** [comment]
translations for this were completed  as part of #440. Testing by schizoduckie and lonelyfarie complete.
closing.

**Comment by lonelyfairie** [comment]
Is there a way to make a two way sync with trakt? I use duckie to download my tv shows, which download to my Plex Server and watch them there, I have a plugin on Plex that automatically pushes the watched show to trakt.tv. So I normally don't mark shows as watched on duckie, but do so directly on trakt, having to manually push shows from trakt to duckie is not optimal so a two way automatic sync would probably be best.

**Comment by garfield69** [comment]
Trakt.TV sync is currently disabled/broken. It's on the to do list for fixing.

**Comment by Js41637** [comment]
What he is saying is if you mark a show as watched on Trakt will it mark it as watched on DuckieTV which is no. I'm not sure if this will be added.

**Comment by lonelyfairie** [comment]
Js41637 Yes, that is what SHE said :P

**Comment by SchizoDuckie** [comment]
@lonelyfairie we don't judge by gender ;)

**Comment by lonelyfairie** [comment]
Good to know, though to be honest working in the tech support industry for 15+ years I am quite use to it by now LOL

**Comment by SchizoDuckie** [comment]
We love all our users equally ^_^

**Comment by lonelyfairie** [comment]
SchizoDuckie, is there a way that when you synch with trakt it only updates the series you haven't hiden in your profile on the website? I mean, I tried the sync and now I realize that it pushed every single series I have marked even 1 episode as watched on trakt and added it to my duckie calendar, which sucks as I have seen many episodes of series I didn't like and abandoned.
Trakt doesn't have a abandoned show feature yet, you can only hide shows from your calendar and profile currently, so I am wondering if there is a way the hidden flag can be respected by duckie when importing series.

I already "cleaned out" duckie but I imagine it would be a problem for users importing for the first time.

**Comment by garfield69** [comment]
Simply from a 'can it be done' perspective, I took a look at the Trakt.TV API, and found that the hidden status of a series is not part of either the 'sync/watched/shows?` and 'sync/collection/shows?' calls we currently do when we perform a trakt.TV import.

I did find a getHiddenItems function. We would need to make two additional calls, `users/hidden/progress_watched?type=show` and `users/hidden/progress_collected?type=show` which would returns lists of hidden shows in each category for us to exclude from importing.
